### PR TITLE
RESKC-204: new nsf cover page 1-6

### PR DESCRIPTION
**Pull request by gathreya**

### CODE REVIEW

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+					break;
+				}
+			}
+		}
+		nsfCoverPage16.setPIInfo(pInfo);
+		nsfCoverPage16.setOtherInfo(otherInfo);
+	}
+
+    protected void setFundingMechanism(NSFCoverPage16 nsfCoverPage16, String answer) {
+        FundingMechanism.Enum fundingMechanism = FundingMechanism.RAPID;
```
Comment (jwillia) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java]: should rapid be the default funding mechanism?  should we leave it blank and have an s2s error if the funding mechanism isn't answered?

Comment (gathreya) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java]: It currently does error ifthe question is not answered. That happens before the code gets to this point.

**Comment by jwillia** [comment]
looks good to me, will let others take a look before merging

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+/**
+ * 
+ * This class is used to generate XML Document object for grants.gov
+ * NSFCoverPageV1.6. This form is generated using XMLBean API's generated by
+ * compiling NSFCoverPageV1.6 schema.
+ * 
+ * @author Kuali Research Administration Team (kualidev@oncourse.iu.edu)
+ */
+@FormGenerator("NSFCoverPageV1_6Generator")
+public class NSFCoverPageV1_6Generator extends NSFCoverPageBaseGenerator {
```
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java]: what is with the 1_6-V1.6?  Should it be NSFCoverPage16V1_6Generator and NSF_CoverPage_1_6-V1.6.xsl?

**Comment by leoherbie** [comment]
Can you please add an integration test in the kc project for this?

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java (hunk 2 above)
Comment (gathreya) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java]: not sure, that seems to be the convention for all cover pages

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+
+		nsfCoverPage16Document.setNSFCoverPage16(nsfCoverPage16);
+		return nsfCoverPage16Document;
+	}
+
+
+
+	private void setFundingOpportunityNumber(NSFCoverPage16 nsfCoverPage16) {
+		if (pdDoc.getDevelopmentProposal().getProgramAnnouncementNumber() != null) {
+			if (pdDoc.getDevelopmentProposal().getProgramAnnouncementNumber()
```
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java]: There is a much simplier way to do this using StringUtils.  I would suggest doing that

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+	 *
+	 * @return AttachedFileDataType[] array of attachments based on the
+	 *         narrative type code.
+	 */
+	private AttachedFileDataType[] getAttachedFileDataTypes() {
+		List<AttachedFileDataType> attachedFileDataTypeList = new ArrayList<>();
+		for (NarrativeContract narrative : pdDoc.getDevelopmentProposal()
+				.getNarratives()) {
+			if (narrative.getNarrativeType().getCode() != null) {
+				int narrativeTypeCode = Integer.parseInt(narrative.getNarrativeType().getCode());
```
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java]: are narrative type codes always ints?

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+		return attachedFileDataTypeList
+				.toArray(new AttachedFileDataType[attachedFileDataTypeList
+						.size()]);
+	}
+
+    protected AttachedFileDataType getDataManagementPlan() {
+        AttachedFileDataType attachedFileDataType = null;
+        for (NarrativeContract narrative : pdDoc.getDevelopmentProposal()
+                .getNarratives()) {
+            if (narrative.getNarrativeType().getCode() != null
```
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java]: are narrative type codes always ints?

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+            }
+        }
+        return attachedFileDataType;
+    }
+
+    protected AttachedFileDataType getMentoringPlan() {
+        AttachedFileDataType attachedFileDataType = null;
+        for (NarrativeContract narrative : pdDoc.getDevelopmentProposal()
+                .getNarratives()) {
+            if (narrative.getNarrativeType().getCode() != null
```
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java]: are narrative type codes always ints?

Comment (gathreya) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java]: yes

File: coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java (hunk 2 above)
Comment (leoherbie) [inline coeus-s2sgen-impl/src/main/java/org/kuali/coeus/s2sgen/impl/generate/support/NSFCoverPageV1_6Generator.java]: ok.  please follow the convention then

**Pull request merged**